sync/src/types/mod.rs

```diff
@@ -216,6 +216,7 @@ pub struct PeerState {
     pub sync_started: bool,
     pub headers_sync_controller: Option<HeadersSyncController>,
     pub peer_flags: PeerFlags,
+    sync_connected: bool,
```

Comment:
I understand the purpose of this field. As it is a bit implicit for normal readers, I suggest adding some comments.